bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdataframe.py

```diff
@@ -436,24 +458,18 @@ def _MakeNumpyDataFrame(np_dict):
     using the keys as column names and the numpy arrays as data.
     """
     import ROOT
-    import cppyy
-    import platform
+
+    reinterpret_cast = ROOT.module.cppyy.ll.reinterpret_cast
 
     if not isinstance(np_dict, dict):
         raise RuntimeError("Object not convertible: Python object is not a dictionary.")
 
     if len(np_dict) == 0:
         raise RuntimeError("Object not convertible: Dictionary is empty.")
 
-    address_prefix = "0x" if platform.system() == "Windows" else ""
-
     pyvecs = dict()
 
-    # Add PyObject (dictionary) holding RVecs to data source
-    code = "ROOT::Internal::RDF::MakeNumpyDataFrame("
-    code += f"reinterpret_cast<PyObject*>({address_prefix}{id(pyvecs)}), "
-
-    def write_vec_code(key, value):
+    def make_name_proxy_pair(key, value):
```

Comment:
Can this be made a module-level private function?

Reply:
Sure!